src/view/renderer.js

```diff
@@ -602,24 +601,33 @@ export default class Renderer {
 	 */
 	_updateDomSelection( domRoot ) {
 		const domSelection = domRoot.ownerDocument.defaultView.getSelection();
-		const oldViewSelection = domSelection && this.domConverter.domSelectionToView( domSelection );
 
-		if ( oldViewSelection && this.selection.isEqual( oldViewSelection ) ) {
-			return;
-		}
-
-		if ( oldViewSelection && areSimilarSelections( oldViewSelection, this.selection ) ) {
-			const data = {
-				oldSelection: oldViewSelection,
-				currentSelection: this.selection
-			};
-
-			log.warn(
-				'renderer-skipped-selection-rendering: The selection was not rendered due to its similarity to the current one.',
-				data
-			);
-
-			return;
+		// Below we will check whether DOM selection needs updating at all.
+		// We need to update DOM selection if either:
+		// * it is at incorrect position, or
+		// * it has changed (when compared to view selection).
+		if ( this.domConverter.isDomSelectionCorrect( domSelection ) ) {
```

Comment:
BTW. I really dislike that you cannot comment on a block of code instead of line on GitHub.